src/ast/ast_visitor.h

```diff
@@ -10,7 +10,7 @@
 
 typedef struct ast_visitor ast_visitor;
 
-typedef bool (*visit)(const cypher_astnode_t *n, bool start, ast_visitor *visitor);
+typedef bool (*visit)(const cypher_astnode_t *n, bool start, ast_visitor *visitor, bool *err);
```

Comment:
I don't like this change because now all visitor methods need to have this in the signature
instead ho having in in the visitor context and not all the uses of the visitor is for validation

Reply:
after talking with @raz-mon the intention of this error is to terminate the visitor so I prefer to specifying by changing the return value from bool to enum like https://clang.llvm.org/doxygen/group__CINDEX__CURSOR__TRAVERSAL.html#ga99a9058656e696b622fbefaf5207d715